parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -215,28 +239,27 @@ extractHyperkube() {
     CLI_TOOL=$1
     path="/home/hyperkube-downloads/${KUBERNETES_VERSION}"
     pullContainerImage $CLI_TOOL ${HYPERKUBE_URL}
-    if [[ "$CLI_TOOL" == "docker" ]]; then
-        mkdir -p "$path"
-        # Check if we can extract kubelet and kubectl directly from hyperkube's binary folder
+    mkdir -p "$path"
+
+    if [[ "$CLI_TOOL" == "ctr" ]]; then    
+        if ctr --namespace k8s.io run --rm --mount type=bind,src=$path,dst=$path,options=bind:rw ${HYPERKUBE_URL} extractTask /bin/bash -c "cp /usr/local/bin/{kubelet,kubectl} $path"; then
+            mv "$path/kubelet" "/usr/local/bin/kubelet-${KUBERNETES_VERSION}"
+            mv "$path/kubectl" "/usr/local/bin/kubectl-${KUBERNETES_VERSION}"
+        else
+            ctr --namespace k8s.io run --rm --mount type=bind,src=$path,dst=$path,options=bind:rw ${HYPERKUBE_URL} extractTask /bin/bash -c "cp /hyperkube $path"
+        fi 
+    
+    else 
```

Comment:
Do we need to support containerd for 1.18 ? 
there is no hyperkube for k8s 1.19+, so we can save this if we don't need to support 1.18 and below versions

Reply:
planning on GA'ing on 1.19+ only. 
I'd like to keep this though so our cse scripts for a few reasons:
1. we still need to support upgrading of existing containerd preview clusters which could be running versions lower than 1.19.x
2. general backwards/forwards compatibility